cdap/http.go

```diff
@@ -44,12 +45,21 @@ func httpCall(client *http.Client, req *http.Request) ([]byte, error) {
 
 	b, err := doHTTPCall(client, req)
 
-	// CDAP REST intermittently returns 500 internal errors, we will retry on 500s once.
+	// CDAP REST intermittently returns 500, 504, etc. internal errors, we will retry on 5xxs.
 	var e *httpError
-	if errors.As(err, &e) && e.code == 500 {
-		log.Print("retrying on intermittent 500 error in 2 seconds")
-		time.Sleep(2 * time.Second)
-		b, err = doHTTPCall(client, req)
+	// poor man's 3x EBO.
+	for i := 0; i < 3; i++ {
+		if errors.As(err, &e) && e.code >= 500 && e.code < 600 {
```

Comment:
It is preferred to make the breakout case in the if and move the rest outside i.e.

if !(errors.As(....)....) {
  break
}

log.Println(...)
time.Sleep(...)